Darkchambers Implementation

Ju-li-us opened the pull request:
Hello,

below you’ll find our implementation! We tried to stay as faithful as possible to the ALE implementation (but there are some minor misalignments described below).

Here a rough description of the core functionality

## Core Functionality

### Items

- **Heart (`ITEM_HEART`)**: Restores player health.  
- **Poison (`ITEM_POISON`)**: Damages player health. *(we used a box scaled version of the skull sprite. not sure about that in the ALE reference game, I think it can easily be confused with a spawner there.)*  
- **Trap (`ITEM_TRAP`)**: Looks like a ladder and inflicts greater health damage.  
- **Strongbox (`ITEM_STRONGBOX`)**: Awards 100 points.  
- **Amber Chalice (`ITEM_AMBER_CHALICE`)**: Awards 500 points.  
- **Amulet (`ITEM_AMULET`)**: Awards 1000 points.  
- **Gold Chalice (`ITEM_GOLD_CHALICE`)**: Awards 3000 points.  
- **Shield (`ITEM_SHIELD`)**: Reduces damage taken.  
- **Gun (`ITEM_GUN`)**: Increases player fire rate.  
- **Bomb (`ITEM_BOMB`)**: Destroys all enemies in an area after picked up and space pressed
- **Key (`ITEM_KEY`)**: Unlocks gated areas or cage doors.  
- **Ladder Up (`ITEM_LADDER_UP`)**: Exit to the next level.  
- **Ladder Down (`ITEM_LADDER_DOWN`)**: Descend to a lower level.  
- **Cage Door (`ITEM_CAGE_DOOR`)**: Gates entry to a 3×3 cage containing a reward.  

> All items are spawned randomly and can also be dropped by spawners or enemies when defeated.  We currently spawn more items and enemies than in the ALE implementation to make testing easier  (in the ALE game it can take >1h of playtime until a bomb is found). Also shooting is currently faster to make sure the player does not die unintendedly (shooting speed can be easily lowered by a constant)


### Map

- The map consists of three hard-coded parts: **left**, **middle**, and **right**.  
- Players can traverse between parts using left/right portals, forming a circular loop (e.g., moving right from the right map wraps to the left map).  
- Each level applies a static offset to the map to create visual variation (7 levels in total so far, in the highest level ladder to moving up won't work).
- Entering a map part triggers enemy respawns within that part.
-Each map part currently has a 3×3 locked cage with a reward inside. This shows up more often than the hidden rooms in the ALE version, again to make testing easier.




We’d love to hear any suggestions or ideas you might have  — your feedback would be greatly appreciated!

Thanks so much,
Julius, Paolo, Jakob

k4ntz committed:
Changing the way we handle mods

PaulSeitz committed:
Refactor mod loading and application in Pong environment

- Updated `load_game_mods` to utilize a two-stage modification pipeline (Controller + Wrapper) for better mod management.
- Introduced `JaxAtariModController` and `JaxAtariModWrapper` for handling internal and post-step mods.
- Created specific mod plugins for Pong, including `LazyEnemyMod`, `RandomEnemyMod`, and `AlwaysZeroScoreMod`.
- Enhanced the Pong environment to support new mod functionalities and improved state management.
- Cleaned up the codebase by removing outdated mod handling logic.

PaulSeitz committed:
[Kangaroo mods]
Adapted the existing kangaroo mods to the new proposal structure

PaulSeitz committed:
[Mods / Core] changes to the mods and core to enable modification of constants

Commit:
Added DarkChambers game

Commit:
Implement Dark Chambers state structure and constants

Commit:
Add DarkChambers constants, state, observation, info structures

Ju-li-us committed:
Add files via upload

added version with debug prints

Ju-li-us committed:
Add files via upload

added integrated version

Ju-li-us committed:
added basic implementation

Ju-li-us committed:
added damage + health bar + shooting

Ju-li-us committed:
implemented more features

Ju-li-us committed:
added spawners

Ju-li-us committed:
docs

Ju-li-us committed:
levels implemented

Ju-li-us committed:
typo

Commit:
enemy movement

Commit:
improved enemy movement

Ju-li-us committed:
implemented shooting wizards + keys

Ju-li-us committed:
test pass succesfully

Commit:
improved enemy movement

Ju-li-us committed:
lives + game over implemented

Commit:
Add DarkChambers demo and sprite assets

Ju-li-us committed:
small changes

Ju-li-us committed:
small changes

Ju-li-us committed:
small changes

Ju-li-us committed:
more shooting by wizards

Ju-li-us committed:
Merge remote-tracking branch 'upstream/master'

Ju-li-us committed:
docs made clearer

Ju-li-us committed:
deleted

Ju-li-us committed:
Merge branch 'dev' into master

Ju-li-us committed:
fixed for pull request

fixed for pull request

Commit:
skull sprite

Commit:
wizard sprite

Commit:
green guy sprite

Commit:
stairs sprite

Commit:
skeleton sprite

Commit:
ghost sprite

Commit:
trapdoor stripe

Commit:
pot and barrel sprite

Commit:
player stripe

Commit:
door sprite

Commit:
apple sprite

Commit:
candle sprite

Commit:
key sprite

Commit:
chain sprite

Ju-li-us committed:
implemented sprites - changed game logic

Ju-li-us committed:
Merge branch 'master' of https://github.com/PaoloAndoni/JAXAtari

Commit:
wall and floor pixel

Ju-li-us committed:
adjusted gameplay

Commit:
improved gameplay

Commit:
Remove core.py and .gitignore from PR

Commit:
Remove forbidden changes to core.py and .gitignore

Commit:
moved npy files

Commit:
cleanup

Ju-li-us committed:
adjusted game landscape

Ju-li-us committed:
fixed upper+lower portals

ahmetzahitak commented:
# Dark Chambers First Submission Review

## 1- ALE Similarity: 2.5/4
- The map layout, the sprites, and the gameplay are similar enough to the original game.
- Free space in the map sometimes feels much more than the original, it could be more like a maze. Also, the original map has many more variations in room designs.
- Health bar is visualized with not only a bar but also a number, which I find helpful.
- Enemies and items are quite similar (sometimes identical) to the original game, but dynamic movements of characters are lacking (movements of heads or footsteps).
- Picked items are just colored blocks instead of detailed sprites, which reduces the understanding.
- When two objects are too close to each other, their black backgrounds can be seen (one's background oveershadows the other). This also happens to level and health texts. (probably due to padding)
- In the original game, if an enemy is close enough to the player, it starts attacking, the player loses some health, but the enemy dies at some point. In this implementation, the enemy never dies and keeps attacking the player until the player dies.
- Enemies changes sprites (and finally die) at every hit, but in the original game, they change sprites at every 2 or 3 hits.
- The level and map changes has no animations or transitions in contrast to the original game.
- There is a bug where sometimes the right corner of the map is not rendered at all (the colored bar is missing). And when the player comes to a position (possibly where the small hole should be (where the map transition happens)), the maps changes rapidly back and forth. I could reproduce this by starting the game, go the level up ladder (I think it's always at the right bottom corner) until level 3, then go to the right corner of the map.
- Overall, the gameplay feels a bit faster and a bit harder than the original game, since enemies always chase the player directly without any random movements.

## 2- Implementation Quality : 3/4
- First of all, I didn't realize any delays or lags during gameplay, so everything seems to be running smoothly (both the rendering and the game logic).
- All of the game mechanics seems to be implemented with jax operations: jax.vmap for enemy step and collision detection, jax.lax.fori_loop and jax.lax.scan for iterations, jax.lax.cond for conditional checks.
- The render function (around 700 lines) and especially step function (around 1600 lines) is quite monolithic and long, which makes it hard to read and understand the code. And it would probably make it hard to develop some new features or modes. But there are also some nested functions inside those functions, so it helps with the readability and probably refactoring the code to make it more modular.

M4GI3R commented:
# Review Darkchambers
#### From the Journey Escape Group
## ALE-Similarity

> How close does the gameplay feel to the original? Is it similar enough so both games offer the same gameplay challenge? Or are they completely different?

This is a bit tricky to evaluate, since (as you mentioned in your PR description) your current implementation is more in a testing state (like a sandbox-mode) and therefore can't be easily compared to the gameplay loop of the ALE-version yet. That's why we just add some differences we found to the ones you already listed in your description.

### Differences listed in the PR-description:

|  | JAX               |      ALE          |
|---------------|-------------------|:-----------------:|
|Enemy and Item Spawn | random |  some predefined enemy spawns <br>at the start + some hardcoded enemies &<br> items. most seem random|
|Spawn Frequency | Spawns more items and enemies than in <br>the ALE implementation to make testing easier |    way less enemies   |
|Shooting| faster -> beam shooting | one bullet shots |

### Differences we found:
|                   | JAX               |      ALE          |
|-------------------|:-----------------:|:-----------------:|
|Image Size|160, 160|210, 160|
|UI-Layout|Inside the play area| underneath the play area|
|Background Color| black | grey|
|Player Sprite | static |walking and turning animation|
|Movement Speed|faster| slower|
|Level Size|level width > screen width|level width == screen width|
|Camera Behavior| player always x,y centered| player only y centered, <br>but players x position changes

### Comments
- Adjusting the Level Size and Camera Behavior will probably bring your implementation a lot closer to ALE in terms of general gameplay feeling

### Bugs we noticed:
|                   |                |
|-------------------|:-----------------:|
| items and enemies are renderd in front of the Level lable | <img width="149" height="185" alt="level_render_bug" src="https://github.com/user-attachments/assets/08228ff8-8f20-4b80-8f3a-6026901d57ba" /> |
| the sprites of the spawner and some other enemies have black borders (instead of transparent) | <img width="85" height="115" alt="Spawner_sprite_transparency" src="https://github.com/user-attachments/assets/af53155c-3db9-4e3c-a749-6d384bc27757" /> |
| Player sometimes overlaps with walls after entering a new level (in particular, entering level 6 in our case). The player is then not able to move. |<img width="795" height="797" alt="bug_3" src="https://github.com/user-attachments/assets/1ad953f7-d094-4462-a098-6abda4e544b8" />|
| In Level 4, the walls on the right side leave a vertical gap. If the player moves into this gap and keeps walking right, enemies and items start respawning every frame at random positions. | <img width="1125" height="1022" alt="bug_4" src="https://github.com/user-attachments/assets/03e7f138-35ff-45b8-85da-ce5ce4d69a86" />|


## Implementation Quality
> Is the code efficient or completely bloated? Did the environment need 50 seconds to load in even though its gameplay loop is simple? Is the code modular or obviously monolithic? Is everything jax native or were there obvious jax antipatterns like excessive sequential processing?

- JAX is well used with a lot of vmaps

- The step function (in particular handle normal) is rather verbose and less readable. Some parts could be written as helper function outside the step function (as described in section 4 "Decompose Logic into Helper Functions" of Design Guidelines) for better readability.  
  - E.g. function enemy_step, code that involves player actions (movement and bullets) into player_step, overlap/collision functions
  - Maybe the collision helper functions can be reused e.g. check_bullet_enemy_collision und check_bullet_spawner_collision; check_all_spawners_for_bullet and check_all_enemies_for_bullet

PaulSeitz committed:
Syncing master to dev to add new kangaroo mods

* Changing the way we handle mods

* Changed Pong

* Refactor mod loading and application in Pong environment

- Updated `load_game_mods` to utilize a two-stage modification pipeline (Controller + Wrapper) for better mod management.
- Introduced `JaxAtariModController` and `JaxAtariModWrapper` for handling internal and post-step mods.
- Created specific mod plugins for Pong, including `LazyEnemyMod`, `RandomEnemyMod`, and `AlwaysZeroScoreMod`.
- Enhanced the Pong environment to support new mod functionalities and improved state management.
- Cleaned up the codebase by removing outdated mod handling logic.

* [Kangaroo mods]
Adapted the existing kangaroo mods to the new proposal structure

* [Mods / Core] changes to the mods and core to enable modification of constants

* Listing the mods when wrong mod given, for convenience

* [mods] removed relative import

* [Mods] added option to bundle mods and mods to replace render functions

* [Mods/Kangaroo]
added render patching support, added examples in kangaroo

* [Freeway] added support to change car colors

* [Mods] adjusted remaining mods to new structure

* [Mods] added a kangaroo mods that move the ladders

small adjustment in the freeway speed mod

* [Mods] added functionality to the PostStepModPlugins, including attribute and sprite overrides and a hook for modifications of the state after reset()

* Fixed freeway mods and added new ones

* Add theenemy score rendered to Freeway

* Fixed flame and inverted ladder positions in kangaroo

* Added man modes on Kangaroo, checked if they are working

* Testing renovate for dependency scanning

* Update renovate config

* Change renovate base branch to 'dev'

* Ignore docs directory with renovate

* Add lots of Kangaroo mods

* Add lots of Kangaroo mods (nearly finished)

* Nearly done with rope mod

* Most new mods added

* Add no_bell and no_fruit

* Make png script from npy

---------

Co-authored-by: Quentin <q.delfosse@gmail.com>
Co-authored-by: Dominik Mandok <dominik@mandok.dev>

PaulSeitz committed:
[Rendering] added support for more than 256 colors in the palette using automatic type switches

PaulSeitz committed:
[Benchmark] added a small performance eval script

PaulSeitz committed:
[Bug] fixed an issue where the TRANSPARENT_ID index was out of bounds in the Palette which led to other colors being filled in through overflows

Ju-li-us committed:
fixed wall bug

Dan041k committed:
game logic and rendering update of FlagCapture and Hangman (#233)

* updated FlagCapture renderer

* FlagCapture fixed score and timer rendering

* FlagCapture fix checking at game start

* FlagCapture added correct sprites

* [FlagCapture] corrected player state sprites and corrected score and timer position

* hangman rendering update and sprites

* letter seperator added and new round state bug fixed

PaulSeitz committed:
Syncing master into dev (#234)

* Changing the way we handle mods

* Changed Pong

* Refactor mod loading and application in Pong environment

- Updated `load_game_mods` to utilize a two-stage modification pipeline (Controller + Wrapper) for better mod management.
- Introduced `JaxAtariModController` and `JaxAtariModWrapper` for handling internal and post-step mods.
- Created specific mod plugins for Pong, including `LazyEnemyMod`, `RandomEnemyMod`, and `AlwaysZeroScoreMod`.
- Enhanced the Pong environment to support new mod functionalities and improved state management.
- Cleaned up the codebase by removing outdated mod handling logic.

* [Kangaroo mods]
Adapted the existing kangaroo mods to the new proposal structure

* [Mods / Core] changes to the mods and core to enable modification of constants

* Listing the mods when wrong mod given, for convenience

* [mods] removed relative import

* [Mods] added option to bundle mods and mods to replace render functions

* [Mods/Kangaroo]
added render patching support, added examples in kangaroo

* [Freeway] added support to change car colors

* [Mods] adjusted remaining mods to new structure

* [Mods] added a kangaroo mods that move the ladders

small adjustment in the freeway speed mod

* [Mods] added functionality to the PostStepModPlugins, including attribute and sprite overrides and a hook for modifications of the state after reset()

* Fixed freeway mods and added new ones

* Add theenemy score rendered to Freeway

* Fixed flame and inverted ladder positions in kangaroo

* Added man modes on Kangaroo, checked if they are working

* Testing renovate for dependency scanning

* Update renovate config

* Change renovate base branch to 'dev'

* Ignore docs directory with renovate

* Add lots of Kangaroo mods

* Add lots of Kangaroo mods (nearly finished)

* Nearly done with rope mod

* Most new mods added

* Add no_bell and no_fruit

* Make png script from npy

* fix tank hitbox

* [Kangaroo] fix movement

* [Kangaroo] additional mods

* [Kangaroo] danger_trap

* [Wrappers] add PixelAndObjectObsWrapper that keeps pytree structure for objects

* relaxing dependencies

* Fixed all problems on SpaceInvaders

* [Bug] fixed an issue where the TRANSPARENT_ID index was out of bounds in the Palette which led to other colors being filled in through overflows

* [Kangaroo] fix falling coconut rendering

* [Refactor] adjusted all environments to replicate ALE action mapping and support for 'full_action_set' in the Atari Wrapper

* [Scripts] Implement corrected action mapping for the play.py 

Added a function to map action constants to indices for environments with ACTION_SET. Updated action retrieval to use this mapping.

* Trying to add mods to Seaquest, mines not working

* [Modifications] constants quick fix

* Trying to add mods to Seaquest, mines not working (only color)

* Does not work with multiple mods

* [Mods] removed unintended catching of manually thrown errors in regards to mod conflicts

* Seaquest mods working

---------

Co-authored-by: Quentin <q.delfosse@gmail.com>
Co-authored-by: Dominik Mandok <dominik@mandok.dev>
Co-authored-by: remunds <raban.emunds@gmail.com>
Co-authored-by: kantoz <kantoz@MacBook-Pro.local>
Co-authored-by: Quentin Delfosse <kantoz@google.com>

PaulSeitz committed:
[Refactor] changed datatypes of all constants to flax.struct.PyTreeNode and all states, observations and infos to flax.struct.dataclass to ensure consistency
- changed various utilities to use the new consistent standard
- fixed videocheckers compilation issues
- renamed hauntedhouse, sirlancelot and humancannonball to ensure consistent naming conventions
- adjusted all mods to use the API of the new consistent datatypes (replace() instead of _replace())
- added tests that check if all mods compile

Note on flax.struct.PyTreeNode: Using struct.field(pytree_node=False, ...) on all fields in the constants is technically optional, but since the constants are static by design, it ensures that JAX handles them as static metadata during compilation.

PaulSeitz committed:
[Wrapper] integrated the native downscaling as an option into the wrappers that support pixel based observations (as described in #232)
- adjusted all environments to explicitly pass the renderer config
- fixed a minor action mapping issue in pong
- Deprecated: the 'make_renderer(..)' function was deprecated to make sure that environment modifications are not circumvented on accident

PaulSeitz committed:
Merge training benchmark scripts into dev (#239)

* Changing the way we handle mods

* Changed Pong

* Refactor mod loading and application in Pong environment

- Updated `load_game_mods` to utilize a two-stage modification pipeline (Controller + Wrapper) for better mod management.
- Introduced `JaxAtariModController` and `JaxAtariModWrapper` for handling internal and post-step mods.
- Created specific mod plugins for Pong, including `LazyEnemyMod`, `RandomEnemyMod`, and `AlwaysZeroScoreMod`.
- Enhanced the Pong environment to support new mod functionalities and improved state management.
- Cleaned up the codebase by removing outdated mod handling logic.

* [Kangaroo mods]
Adapted the existing kangaroo mods to the new proposal structure

* [Mods / Core] changes to the mods and core to enable modification of constants

* Listing the mods when wrong mod given, for convenience

* [mods] removed relative import

* add cleanrl benchmarking scripts

* [Mods] added option to bundle mods and mods to replace render functions

* quick test

* [Mods/Kangaroo]
added render patching support, added examples in kangaroo

* fixes freeway cleanRL training

* fix logging when reward clipping

* [Freeway] added support to change car colors

* [Mods] adjusted remaining mods to new structure

* [Mods] added a kangaroo mods that move the ladders

small adjustment in the freeway speed mod

* [Mods] added functionality to the PostStepModPlugins, including attribute and sprite overrides and a hook for modifications of the state after reset()

* Fixed freeway mods and added new ones

* Add theenemy score rendered to Freeway

* Fixed flame and inverted ladder positions in kangaroo

* Added man modes on Kangaroo, checked if they are working

* compatibility for benchmarks

* Testing renovate for dependency scanning

* Update renovate config

* Change renovate base branch to 'dev'

* Ignore docs directory with renovate

* Add lots of Kangaroo mods

* Add lots of Kangaroo mods (nearly finished)

* seems to be working well (compared to ALE cleanRL PPO)

* Nearly done with rope mod

* Most new mods added

* Add no_bell and no_fruit

* Make png script from npy

* fix tank hitbox

* [Kangaroo] fix movement

* [Kangaroo] additional mods

* [Kangaroo] danger_trap

* [Wrappers] add PixelAndObjectObsWrapper that keeps pytree structure for objects

* relaxing dependencies

* Fixed all problems on SpaceInvaders

* [Bug] fixed an issue where the TRANSPARENT_ID index was out of bounds in the Palette which led to other colors being filled in through overflows

* u

* add extra training + fixes

* [Kangaroo] fix falling coconut rendering

* u

---------

Co-authored-by: Quentin <q.delfosse@gmail.com>
Co-authored-by: remunds <raban.emunds@gmail.com>
Co-authored-by: Dominik Mandok <dominik@mandok.dev>
Co-authored-by: kantoz <kantoz@MacBook-Pro.local>

k4ntz committed:
Add citation section to README

Added citation information for JAXAtari framework.

PaulSeitz committed:
[Benchmarks] small adjustments to the training scripts to adjust for recent changes

PaulSeitz committed:
[Benchmarking] small changes to training scripts / configs and addition of new visualisation tools
- added an agent visualisation script
- fixed a bug in the core when loading envs without registered mods
- small adjustment to the seaquest rewards to get them closer to the ALE baseline
- adjustment of asterix observations, removed large negative numbers
- added two new configs that can be used for fast, local pqn training that can (for a small number of environments) reach similar scores to the ALE PPO baseline

PaulSeitz committed:
[Script] added a trajectory script that can save states of a gameplay run of an environment to help in safe refactoring / modding of environments

PaulSeitz committed:
[Tests] trying to let the github actions download the necessary sprites

remunds committed:
Delete game sprites folder

Dan041k committed:
[Riverraid] fixed bullet x position

Dan041k committed:
[Riverraid] fixed rendering bug of objects on UI elements

Dan041k committed:
[Riverraid] spawning objects and enemies earlier and let more fuel spawn

Dan041k committed:
minor change

Dan041k committed:
[Riverraid] adjusted spawning, movement and land collision conditions of enemies

remunds committed:
Venture and Airraid are ready

remunds committed:
[Airraid] fix sprite location

remunds committed:
Add throughput analysis

remunds committed:
u

remunds committed:
fix envpool

felicia-b committed:
Adds amidar

remunds committed:
[Amidar] change maze file location

remunds committed:
[Amidar] adapt to new structure

Commit:
enemy movement v1

Ju-li-us committed:
staged fix

Ju-li-us committed:
Merge branch 'master' of https://github.com/PaoloAndoni/JAXAtari

remunds committed:
[Amidar] fix downscaled rendering

remunds committed:
[Amidar] fix rendering2

remunds committed:
[Amidar] fix constants bug

Ju-li-us committed:
fixed map + some more bugs + color

Ju-li-us committed:
implemented first 3 mods (speed-, poison-, health-restorage-items)

Commit:
added animations

Commit:
added missing sprite

Ju-li-us committed:
fixed wall + enemy rundering bug

k4ntz committed:
Update games_covered.md

Improved visualization and added total

Ju-li-us committed:
fixed potions always spawned bug

Commit:
Add Hammer item with limited uses

Ju-li-us committed:
bugfix + more mods

Commit:
Resolve stash pop conflict in jax_darkchambers

Commit:
current state

Commit:
ale similarity adjustments

Commit:
fix idle corner

jakobsteinke committed:
yaml config

Ju-li-us committed:
fixed no score bug for last enemy

Ju-li-us committed:
no traps

Ju-li-us committed:
implemented checkpoints

Ju-li-us committed:
traps back in

Ju-li-us committed:
walls

Ju-li-us committed:
walls

Commit:
fixed enemey overlap bug

Commit:
fix stairs

Commit:
fixed enemy exit bug

Commit:
updated bullet sprite

Commit:
fixed smoothness error

Commit:
fixed zombie animation

Commit:
fix skel animation

Ju-li-us committed:
mods fixed

Ju-li-us committed:
spawn spoint fix

Ju-li-us committed:
map change

Ju-li-us committed:
less items

Ju-li-us committed:
ornaments

Ju-li-us committed:
ornaments2

Ju-li-us committed:
named tuple fix observation space fix and another wall

jakobsteinke committed:
object centered changes

jakobsteinke committed:
eval configs

Ju-li-us committed:
configs liek ex report

Ju-li-us committed:
typo

jakobsteinke committed:
minus sign fix

jakobsteinke committed:
syntax fix

jakobsteinke committed:
added models folder

Ju-li-us committed:
bugfix bullets

Ju-li-us committed:
adjustments

Ju-li-us committed:
bullet speed changed

Ju-li-us committed:
mod adjusted directly spawned at player

Ju-li-us committed:
adj

Commit:
adjusted movement speed

jakobsteinke committed:
fix pixel execution bug

Ju-li-us committed:
junit test fix

Ju-li-us committed:
fix

Ju-li-us committed:
Merge branch 'master' of https://github.com/PaoloAndoni/JAXAtari

Ju-li-us committed:
action space changes not allowed

Ju-li-us committed:
hammer fix

Ju-li-us committed:
fix

Ju-li-us committed:
Merge remote-tracking branch 'upstream/dev'

Ju-li-us committed:
adjusted

Commit:
grim reaper animation

Commit:
pistol and intra level checkpoints

Commit:
fix wizard sprite

Ju-li-us committed:
deleted for pull request

Ju-li-us committed:
cleaned for pull request

Ju-li-us committed:
fix

Ju-li-us committed:
mods fixed

Ju-li-us committed:
fix

Ju-li-us committed:
mod fix

jakobsteinke committed:
more results

Commit:
better sprites

Ju-li-us committed:
deleted files

Ju-li-us committed:
changes

Ju-li-us committed:
cleaned

Ju-li-us committed:
cleaned

Ju-li-us committed:
synced

Ju-li-us committed:
Remove accidentally committed __pycache__, .DS_Store, and .pyc files

Ju-li-us committed:
fix

Ju-li-us committed:
Clean cache files (__pycache__, pyc, DS_Store)

Ju-li-us committed:
fix

Ju-li-us committed:
cleaned

Commit:
fix test timeout

Commit:
better memory usage

Ju-li-us committed:
cleaned

Ju-li-us committed:
moved spawner

Commit:
z index fix

Commit:
better door

Commit:
cleanup

Ju-li-us committed:
trapdoor bugfix

Ju-li-us committed:
Merge branch 'master' of https://github.com/PaoloAndoni/JAXAtari

Commit:
improve stairs
